Source/Core/Core/PowerPC/Jit64/Jit.cpp

```diff
@@ -355,9 +355,14 @@ bool Jit64::Cleanup()
 
   if (jo.optimizeGatherPipe && js.fifoBytesSinceCheck > 0)
   {
+    MOV(64, R(RSCRATCH), PPCSTATE(gather_pipe_ptr));
+    SUB(64, R(RSCRATCH), PPCSTATE(gather_pipe_base_ptr));
```

Comment:
We can optimize this further: If we pack the GP data into the ppcState struct as well, this will be a CMP to the ppcState register instead of two loads.

Reply:
alternatively:

    MOV(64, R(RSCRATCH), PPCSTATE(gather_pipe_base_ptr));
    ADD(64, R(RSCRATCH), Imm32(GPFifo::GATHER_PIPE_SIZE))
    CMP(64, PPCSTATE(gather_pipe_ptr), R(SCRATCH));
    FixupBranch exit = J_CC(CC_BE);

or:

    MOV(64, R(RSCRATCH), Imm64(PowerPC::ppcState.gather_pipe_base_ptr + GPFifo::GATHER_PIPE_SIZE));
    CMP(64, PPCSTATE(gather_pipe_ptr), R(SCRATCH));
    FixupBranch exit = J_CC(CC_BE);